src/tensor/cuda/allocate.rs

```diff
@@ -138,6 +138,16 @@ where
     }
 }
 
+impl<E: Unit> TensorFromVec<E> for Cuda {
+    fn try_tensor_with_shape<S: Shape>(
+        &self,
+        src: Vec<E>,
+        shape: S,
+    ) -> Result<Tensor<S, E, Self>, Self::Err> {
+        self.take_cpu_tensor(self.cpu.try_tensor_with_shape(src, shape)?)
```

Comment:
Let's fill in this method directly instead of going through cpu here. Mainly because we can use self.dev.take_async(src) directly since we have the vec, which will be much faster than going through the cpu!

It's fine to duplicate error handling here IMO.